### ISSUE TITLE
Can't cope with a y-combinator.

### ISSUE THREAD

**Issue by Incognito** [toolness/slowmo-js#7]
http://toolness.github.com/slowmo-js/?code=(function%20(ycomb)%20%7B%0A%20%20%20%20return%20(function(n)%7B%0A%20%20%20%20%20%20%20%20return%20ycomb(ycomb)(n)%3B%0A%20%20%20%20%7D)%3B%0A%7D)(function%20(self)%20%7B%0A%20%0A%20%20%20%20return%20function%20(x)%20%7B%0A%20%20%20%20%20%20%20%20if%20(%20x%20%3E%200%20)%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20console.log(x)%3B%0A%20%20%20%20%20%20%20%20%20%20%20%20return%20self(self)(x-1)%3B%0A%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%7D%3B%0A%7D)(5)%3B&filterrange=244-244

**Comment by toolness** [comment]
Hey, I think that this was fixed by bc3ecf5b07d09bac6ab00b0433a6c221315db137. I didn't realize that a y-combinator was something other than an American seed-stage startup funding firm, and right now I'm a bit too tired to find out exactly what it's supposed to do, but running your code:

``` javascript
(function (ycomb) {
    return (function(n){
        return ycomb(ycomb)(n);
    });
})(function (self) {

    return function (x) {
        if ( x > 0 ) {
            console.log(x);
            return self(self)(x-1);
        }
    };
})(5)
```

now logs `5 4 3 2 1` as five separate console messages and returns undefined, which is what this snippet of code seems to do in node, so I am guessing the bug is fixed now. Feel free to re-open this issue if I'm wrong, though, and thanks for the report.

**Comment by Incognito** [comment]
Thanks for the support!

You can think of a y-combinator as a way to do recursion calls when the calling function doesn't have a reference to its self (which was the case in older languages, and some really deep corner cases in JS).
